lib/LoginStrategy.ts

```diff
@@ -406,4 +413,38 @@ export class LoginStrategy extends BaseStrategy {
 
     return true;
   }
+
+  /**
+   * Filters verified roles into valid and invalid roles and sets validation status
+   * @param userRoles verified user role credential
+   * @returns
+   */
+  private validateAcceptedRoles(userRoles: RoleStatus[]): {
+    userRoles: RoleStatus[];
+    authorisationStatus: boolean;
+  } {
+    let authorisationStatus = true;
+
+    if (this.acceptedRoles.size > 0) {
+      this.acceptedRoles.forEach((role) => {
+        const userRole = userRoles.find(
+          (verifiedRole) => verifiedRole.namespace === role
+        );
+        if (!userRole) {
+          userRoles.push({
+            name: role,
+            namespace: role,
+            error: 'Role credential not found for the user.',
+          });
+          authorisationStatus = false;
+        } else if (userRole.error) {
+          authorisationStatus = false;
+        }
+      });
+    }
```

Comment:
@nichonien Is this is a change in the behaviour from before?
I think the previous code (below) passed if **any** of the roles from `acceptedRoles` where verified and present. I think the new code requires that  **all** accepted roles are present?
If this is indeed a change, was this change intended?
```
this.acceptedRoles.size > 0 &&
!uniqueRoles.some(({ namespace }) => {
          return this.acceptedRoles.has(namespace);
})
```

Reply:
@jrhender No, this is a mistake. I will correct it.

Reply:
Ok, great. Maybe we can add a test for this case (should be authorized when only one of the acceptedRoles is verified and other is not verified) either in this PR or a separate one?

Reply:
Adding a test is not possible, since mocking `ClaimVerifier` is not feasible. Our earlier discussion - https://github.com/energywebfoundation/passport-did-auth/pull/331#discussion_r985457467